pkg/controller/operators/olm/operator.go

```diff
@@ -158,6 +187,80 @@ func (a *Operator) syncDeployment(obj interface{}) (syncError error) {
 	return nil
 }
 
+func (a *Operator) updateDeploymentAnnotation(op *v1alpha2.OperatorGroup) error {
+	selector, err := metav1.LabelSelectorAsSelector(&op.Spec.Selector)
+	if err != nil {
+		return err
+	}
+	operatorGroupOpts := metav1.ListOptions{LabelSelector: selector.String()}
+	// TODO(jpeeler): this needs to use user impersonation with op.Spec.ServiceAccount
+	namespaceList, err := a.OpClient.KubernetesInterface().CoreV1().Namespaces().List(operatorGroupOpts)
+	if err != nil {
+		return err
+	}
+
+	nsCount := len(namespaceList.Items)
+
+	if len(op.Status.Namespaces) == nsCount {
+		for i, v := range namespaceList.Items {
```

Comment:
I think it would be better to implement this as a set (`map[string]struct{}{}`) instead of a list; I'm not sure if we get order guarantees from `List`